rust/libceed/src/vector.rs

```diff
@@ -348,11 +348,7 @@ impl<'a> Vector<'a> {
     // Error handling
     #[doc(hidden)]
     fn check_error(&self, ierr: i32) -> crate::Result<i32> {
-        let mut ptr = std::ptr::null_mut();
-        unsafe {
-            bind_ceed::CeedVectorGetCeed(self.ptr, &mut ptr);
-        }
-        crate::check_error(ptr, ierr)
+        unsafe { crate::check_error(bind_ceed::CeedVectorReturnCeed(self.ptr), ierr) }
     }
```

Comment:
As a matter of style/ergonomics, we can make a safe private function `Vector::ceed(&self) -> Ceed`.

We could also make `crate::check_error` accept a closure so we aren't eagerly making real calls just to obtain a `Ceed` we won't use. Then this function would look like `crate::check_error(|| self.ceed(), ierr)`. That seems more in the spirit of standard library interfaces like [`ok_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else)

The ergonomics above are perhaps kinda moot in that I think the unsafe FFI blocks should use a closure, perhaps similar to RSMPI `with_uninitialized` (which provisions a suitable pointer type for an output argument and does not currently do error checking, but could be extended for that).

Reply:
Something like 11544396610b36de1cb2f0d18032eefe5c670568?

Reply:
Yup, exactly. Feels nicer to have less/narrower `unsafe`.